doc/jointcal.tex

```diff
@@ -431,27 +465,47 @@ \subsection{Minimization approach}
 $$
 In the code, we take advantage of the fact that each term of the
 $\chi^2$ only depends on a small number of parameters. The data
-stuctures allow us to collect easily the indices of these parameters,
+stuctures allow us to rapidly collect the indices of these parameters,
 and we evaluate in fact the $H$ matrices for these indices only.
 
-The computation of the Jacobian and the gradient is orchestrated in
-the \ClName{AstromFit} class.  The routines
-\RoutineName{AstromFit::LSDerivatives1} and
-\RoutineName{AstromFit::LSDerivatives2} evaluate the contributions to
+The computation of the Jacobian and the gradient is performed in
+the \ClName{AstrometryFit} class. The methods
+\RoutineName{AstrometryFit::leastSquareDerivativesMeasurement} and \\
+\RoutineName{AstrometryFit::leastSquareDerivativesReference} evaluate the contributions to
 the Jacobian and gradient of the $\chi^2$ from the measurement terms
 and the references terms respectively. In these routines, the Jacobian
 is represented as a list of triplets $(i,j,J_{ij})$ describing its
 elements. This list is then transformed into a representation of
 sparse matrices suitable for algebra, and in particular suitable to
 evaluate the product $J^TJ$. Once we have evaluated $H\equiv J^TJ$, we
-can solve $HX=-g$ using a Cholesky factorization. For sparse linear
-algebra, at least the Cholmod and Eigen packages provide the required
-functionalities. It turns out that for practical problems, the
-factorization is the most CPU intensive part of the calculations, and
+can solve $HX=-g$ (see eq. \ref{eq:step_definition}, where the Hessian, $H$, is as
+defined in \ref{eq:def_hessian}, $X$ is the vector of parameter/fittedStar deltas
+we are solving for, and $g$ is the gradient of the $\chi^2$, given in \ref{eq:gradient})
+using a Cholesky factorization. For sparse linear
+algebra, the Cholmod and Eigen packages provide the required
+functionality. It turns out that for practical problems, the
+calculation of $J^TJ$ or the factorization are the most CPU intensive
+parts of the calculations, and
```

Comment:
I think it depends a lot on the models, but we'll have to see (e.g., our derivative approach saves us some work, I think). We'll definitely have to spend some time profiling jointcal in the future: I've reshuffled a lot of things, and haven't always been totally performance oriented (i.e., just make it work) in my thinking.